### ISSUE TITLE
[nullsafety] closures with uninitialized local vars

### ISSUE THREAD

**Issue by Gama11** [HaxeFoundation/haxe#7803]
The following code isn't null-safe, but null safety doesn't complain:

```haxe
@:nullSafety
class Main {
	static function main() {
		var a:Array<Int>;
		function f() {
			a.push(0);
		}
		f();
	}
}
```

>TypeError: Cannot read property 'push' of undefined

**Comment by back2dos** [comment]
Shouldn't this be handled by the compiler's check against uninitialised variables?

**Comment by Gama11** [comment]
That doesn't seem to apply for locals used in closures (but perhaps ideally it should).

**Comment by nadako** [comment]
Concerning null-safety this looks okay, since `a` is not `Null<Array<Int>>` so it's really about the unitialized vars. I wonder though, if the null-safety checker will complain if `a` is typed as `Null<Array<Int>>`.

**Comment by Simn** [comment]
#7447

**Comment by RealyUniqueName** [comment]
Looks like a compiler's job. I don't think null safety should check this. Especially after #7447.

> I wonder though, if the null-safety checker will complain if `a` is typed as `Null<Array<Int>>`

It will.

**Comment by RealyUniqueName** [comment]
Closing in favor of #7447
